src/signal.rs

```diff
@@ -408,13 +408,19 @@ mod tests {
     use std::time::Duration;
 
     static mut SIGNAL_HANDLER_CALLED: bool = false;
+    static mut SIGNAL_HANDLER_CALLED_FLAG: bool = false;
```

Comment:
Having these two global variables with similar names (`SIGNAL_HANDLER_CALLED` and `SIGNAL_HANDLER_CALLED_FLAG`), and also the two functions with similar names (`handle_signal_func` and `handle_signal`) is a bit confusing, and might not scale well when we keep adding tests.

We can replace the multiple global variables with an array of `bool`s (or `Atomic<Something>` if we want to get rid of the unsafe), basically one for each signal number. Then we can keep a single handler function which knows the index corresponding to the current signal by inspecting the first parameter it receives (I'm assuming it contains the signal number).

Reply:
Thanks for the reviewing.
I found it is hard to define handler flags arrary/vector with SIGRTMAX() capacity, because we'd like this array as static, but calls in statics are limited to constant functions. Maybe we can only define a simple array with two members for tests usage.